Feature/IGDBGameService

PS2emu opened the pull request:
IGDBGameService is able to translate from Twitch ID to IGDB ID. Note, the URL building could be torn out and potentially made in its own method, along with replacing the fields with enums as referenced in #8 .

Also, for some reason Github states this PR cannot be merged, but your branch is only 1 commit behind mine. I'm unsure why this is happening.

PS2emu committed:
Basic IGDBAPI integration

PS2emu committed:
Fixed my Maven issues. Uncommented from tests

PS2emu committed:
Created IGDBGameService. Links Twitch and IGDB

PS2emu committed:
Merge branch 'feature/igdb-integration' into feature/new-igdb-service

PS2emu commented:
The merge conflict was resolved by deleting unnecessary comments.`retrieveGameById` had been commented out and moved from IGDBAPI to IGDBGameService.

victorwrotethis commented on src/main/java/net/gecore/streamerendpoints/service/igdb/IGDBGameService.java:
```diff
@@ -0,0 +1,41 @@
+    private final IGDBAPI igdbapi;
+    private final TwitchConfig twitchConfig;
+    private final AuthService authService;
+//    private final Logger LOGGER = LoggerFactory.getLogger(IGDBService.class);
```
You want to avoid pushing commented out code.

PS2emu replied:
It's been commented out because I'm unsure how you want me to use the Logger library. For example, you've set the logger in `TwitchAPI.java` to `TwitchAPI.class`. However, in `TwitchGameService.java` you've set it to `StreamService.class`. How would you like this used in IGDBGameService?

victorwrotethis replied:
True, that's definitely a copy paste mistake just like here. Still remove it for now, there's almost never a reason to commit commented code especially in a PR. It's just a bad practice. _Might be_ is a good way to have a lot of lines doing nothing.

Logging makes more sense once the entire flow is ready.

victorwrotethis commented on src/main/java/net/gecore/streamerendpoints/service/igdb/IGDBGameService.java:
```diff
@@ -0,0 +1,41 @@
+    public Integer externalGameID(String externalId)
+            throws MalformedURLException, TwitchAPIException, IGDBAPIException {
+        URL url = new URL("https://api.igdb.com/v4/external_games");
+        String body = "fields game; where category = 14 & uid = \"" + externalId + "\";";
```
It might be a better idea to elevate the `14` assignment to a variable to help indicate it is the Steam category. That way you immediately know without having to refer to the API-docs. In the future it can be an enum, but that would be overkill right now being that we only use 1 of them.

You also might want to use String.format() for building the body as if there's more than 1 variable it tends to get messy.

victorwrotethis commented on src/main/java/net/gecore/streamerendpoints/service/igdb/IGDBGameService.java:
```diff
@@ -0,0 +1,41 @@
+
+    public String retrieveGameById(Integer gameId)
+            throws IGDBAPIException, TwitchAPIException, MalformedURLException {
+        URL url = new URL("https://api.igdb.com/v4/games");
```
I'd rather already see in this pull request that the url buildup is similar to the ones used in the twitch specific services. For now simply turning the last part of the URL, `games` and `external games`, into values of an 'IGDBEndpoint' enum should work.

PS2emu replied:
Would you rather I also make a URLHelper class in the `igdb` package, or better yet add a `buildIGDBUrl` method to `URLHelper` and move it to the `utils` package?

victorwrotethis replied:
Yeah I was going to do the latter afterwards, but you can do it too. Make sure to rename the old method to buildTwitchUrl.

PS2emu committed:
Created IGDB endpoint Enums

PS2emu committed:
Created buildIGDBUrl, modified buildTwitchUrl, moved to Utils

PS2emu committed:
URLHelper.buildUrl -> URLHelper.buildTwitchUrl

PS2emu committed:
PR Code Review Changes

Unnecessary comment removed.
Clarified IGDB category query param.
Cleaner string building process.

PS2emu committed:
Merge branch 'feature/new-igdb-service' of https://github.com/PS2emu/streamer-endpoints into feature/new-igdb-service

Edits made in Github PR. Need to update to that single commit.

PS2emu committed:
Created "fields" enums for "games" endpoint.

PS2emu commented:
With this latest commit I believe #8 can be closed by this PR.
